### PR TITLE
refactor: use new da-codec interfaces

### PR DESCRIPTION
**Pull request by colinlyguo**
## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [x] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [x] This PR is not a breaking change

## Release Notes

- **New Features**
	- Introduced a new key for tracking the last synced L1 block number for rollup events.
	- Enhanced handling of commit batch data with codec versioning in the `CalldataBlobSource`.

- **Bug Fixes**
	- Simplified the `getCommittedBatchMeta` method to improve clarity and reduce complexity.

- **Refactor**
	- Streamlined batch type handling in the `BatchQueue`.
	- Consolidated codec handling in the `RollupSyncService`, removing unnecessary complexity.

- **Chores**
	- Updated module dependencies to the latest versions.

### CODE REVIEW

**Review by jonastheis — DISMISSED** [review]

File: rollup/da_syncer/da/commitV0.go [hunk 1]
```diff
@@ -129,28 +130,36 @@ func (c *CommitBatchDAV0) Blocks() []*PartialBlock {
...
+		quo := index / 256
+		rem := index % 256
+		return skippedBitmap[quo].Bit(int(rem)) == 1
+	}
+
 	// get all necessary l1 messages without skipped
 	currentIndex := parentTotalL1MessagePopped
 	for index := 0; index < totalL1MessagePopped; index++ {
-		if encoding.IsL1MessageSkipped(decodedSkippedBitmap, currentIndex-parentTotalL1MessagePopped) {
+		if isL1MessageSkipped(decodedSkippedBitmap, currentIndex-parentTotalL1MessagePopped) {
```
Comment (colinlyguo) [inline rollup/da_syncer/da/commitV0.go]: @NazariiDenha do you think we should add this function in da-codec repo?

Comment (NazariiDenha) [reply rollup/da_syncer/da/commitV0.go]: Previously it was there and I think it was better

Comment (colinlyguo) [reply rollup/da_syncer/da/commitV0.go]: OK.

Comment (colinlyguo) [reply rollup/da_syncer/da/commitV0.go]: done in [bbd7b73](https://github.com/scroll-tech/go-ethereum/pull/1068/commits/bbd7b73b9bdbe4eb500aed604002befc8d16664c).

**Review by jonastheis — DISMISSED** [review]

**Comment by georgehao** [comment]
@coderabbitai review

**Review by georgehao — APPROVED** [review]

**Pull request merged**